File: 브루트 포스/1018/1018.py
n, m = map(int, input().split())
answer = 999999999
arr = [None] * n

for i in range(n):
   s = input()
   arr[i] = [None] * m
   for j in range(m):
      if s[j] == 'B':
         arr[i][j] = 0
      else:
         arr[i][j] = 1

for i in range(0, n - 7):
   for j in range(0, m - 7):
      answer1, answer2 = 0, 0
      for k in range(i, i + 8):
         for l in range(j, j + 8):
            answer1 += arr[k][l] ^ ((k + l) % 2)
            answer2 += arr[k][l] ^ ((k + l + 1) % 2)
      answer = min(answer, answer1, answer2)
print(answer)

File: 브루트 포스/1018/1018.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main()
{
   int n, m;
   int answer = 999999999;
   string s;   
   cin>>n>>m;
   int arr[n][m];

   for (int i = 0; i < n; ++i)
   {
      cin>>s;
      for(int j = 0; j < m; ++j)
      {
         if (s[j] == 'B')
            arr[i][j] = 0;
         else
            arr[i][j] = 1;
      }
   }

   for (int i = 0; i < n - 7; ++i)
      for (int j = 0; j < m - 7; ++j)
      {
         int answer1 = 0, answer2 = 0;
         for (int k = i; k < i + 8; ++k)
            for (int l = j; l < j + 8; ++l)
            {
               answer1 += arr[k][l] ^ ((k + l) % 2);
               answer2 += arr[k][l] ^ ((k + l + 1) % 2);
            }
         answer = min(min(answer1, answer2), answer);
      }
   cout<<answer;

   return 0;
}